package/android/src/main/java/com/mrousavy/camera/core/CameraSession+Video.kt

```
@@ -108,3 +127,17 @@ fun CameraSession.resumeRecording() {
   val recording = recording ?: throw NoRecordingInProgressError()
   recording.resume()
 }
+
+fun CameraSession.startEndOfStream() {
+  setEISMode("2");
+}
+
+fun CameraSession.restoreEndOfStream() {
+  setEISMode(Settings.Global.getString(context.contentResolver, "enable_EIS"));
+}
+
+fun CameraSession.setEISMode(newMode: String) {
+  Log.i(CameraSession.TAG, "EIS: changed from: " + SystemProperties.get("persist.vendor.camera.enableEIS"));
+  SystemProperties.set("persist.vendor.camera.enableEIS", newMode);
```

I believe this whole method needs to be wrapped in a try/catch in the case it's running on an earlier version of the firmware (that doesn't have the SELinux rules allowing us to access this system property)

---

Ahhh nice one - good catch.

I've put in a check now. I don't think there's any issue with just failing silently since toggling that bit would have no effect anyways but let me know if I've thought that through wrong